functions/bullet_count.h

```diff
@@ -1,7 +1,7 @@
 #ifndef FUNCTIONS_BULLETCOUNT_H
 #define FUNCTIONS_BULLETCOUNT_H
 
-#ifdef BLASTER_SHOTS_UNTIL_EMPTY
+#ifdef PROP_HAS_BULLET_COUNT
```

Comment:
oh same issue.
Is it a crime if we just moved 
```cpp
#include "functions/bullet_count.h"
#include "functions/blaster_mode.h"
```
down under the `PROP_TYPE prop;` line?
They're up with the rest of the functions nice and orderly, but they do pertain to a prop which is different than the blade style stuff. Then the definitions could move right into the prop like you had though originally?

Reply:
If you move them, then the style won't be available in CONFIG_PRESETS.

Reply:
hmm. not sure what's best to do then. I was just trying to address your concern "not all blasters have bullet counts" so using BULLET_COUNT_UNTIL_EMPTY won't always work.

Reply:
BULLET_COUNT_UNTIL_EMPTY is the define that specifies if a blaster has bullet counts or not.
It is the right define *for bullet counts*, it is not the right define for blaster modes.

Reply:
oh yeah agreed. Now that I look, what you had said was "Not all blasters have limited number of shots." So meaning no limit. I guess then the bullet count would be an increasing number, not a "remaining" number?
Either way , the actual define is `BLASTER_SHOTS_UNTIL_EMPTY`. 
So are you thinking a new define that simply enables bullet counts like `BLASTER_HAS_BULLET_COUNT`?
I guess that's what i was leaning toward with the PR here using `PROP_HAS_BULLET_COUNT`.

Words are hard.

Reply:
The blaster prop has a counter that starts at BLASTER_SHOTS_UNTIL_EMPTY and then counts down.
When you hit zero, you need to reload.
If BLASTER_SHOTS_UNTIL_EMPTY is undefined, it sets it to -1, and you never need to reload, so bullet counts are meaningless, which is why it's fine to not have any of these style templates defined unless BLASTER_SHOTS_UNTIL_EMPTY was defined.

If there was no define I could use, I would do it the same way that we did with the blaster mode. Declare a function, then only define the function if PROP_HAS_BULLET_COUNT was defined. This would lead to a linker error if you tried to use these styles on something that wasn't a blaster. However, the linker error is a bit harder to understand, so using a define is probably better, if it is available. (Which it is in this case.)